metriq_gym/benchmarks/clops.py

```diff
@@ -0,0 +1,155 @@
+                for pn in prospective_neighbors:
+                    if pn not in qubit_map and pn not in faulty_qubits:
+                        new_neighbors.append(pn)
+                qubit_map = qubit_map + new_neighbors
+                prospective_neighbors = []
+                for nn in new_neighbors:
+                    potential_neighbors = list(cm.neighbors(nn))
+                    for pn in potential_neighbors:
+                        if pn not in prospective_neighbors:
+                            prospective_neighbors.append(pn)
+                new_neighbors = []
+                if len(qubit_map) >= width:
+                    return qubit_map[:width]
+    raise ValueError("Insufficient connected qubits to create set of %d qubits", width)
+
+
+# adapted from submodules/qiskit-device-benchmarking/qiskit_device_benchmarking/clops/clops_benchmark.py::prepare_clops_circuits
+# As opposed to the original version of this function, this takes a topology graph as an argument instead of an IBM's coupling map object.
+def append_2q_layer(
+    qc: QuantumCircuit, topology_graph: rx.PyGraph, basis_gates: list[str], rng
+) -> None:
+    """
+    Add a layer of random 2q gates.
+    """
+    available_edges = set(topology_graph.edge_list())
+    while len(available_edges) > 0:
+        edge = tuple(rng.choice(list(available_edges)))
+        available_edges.remove(edge)
+        edges_to_delete = []
+        for ce in list(available_edges):
+            if (edge[0] in ce) or (edge[1] in ce):
+                edges_to_delete.append(ce)
+        available_edges.difference_update(set(edges_to_delete))
+        if "ecr" in basis_gates:
+            qc.ecr(*edge)
+        elif "cz" in basis_gates:
+            qc.cz(*edge)
+        else:
+            qc.cx(*edge)
+    return
+
+
+# adapted from submodules/qiskit-device-benchmarking/qiskit_device_benchmarking/clops/clops_benchmark.py::prepare_clops_circuits
+def prepare_clops_circuits(
+    width: int,
+    layers: int,
+    num_circuits: int,
+    basis_gates,
+    topology_graph: rx.PyGraph,
+    total_qubits: Optional[int],
+    seed: int = 0,
+) -> list[QuantumCircuit]:
+    qubit_map = create_qubit_map(width, topology_graph, faulty_qubits=[], total_qubits=total_qubits)
+
+    for edge in topology_graph.edge_list():
+        if edge[0] not in qubit_map or edge[1] not in qubit_map:
+            topology_graph.remove_edge(*edge)
+
+    qc = QuantumCircuit(max(qubit_map) + 1, max(qubit_map) + 1)
+    qubits = [qc.qubits[i] for i in qubit_map]
+
+    parameters = []
+    rng = np.random.default_rng(seed)
+    for d in range(layers):
+        append_2q_layer(qc, topology_graph, basis_gates, rng)
+        parameters += append_1q_layer(qc, qubits, parameterized=True, parameter_prefix="L" + str(d))
+
+    for idx in range(width):
+        qc.measure(qubit_map[idx], idx)
+
+    # Parameters are instantiated for each circuit with random values
+    parametrized_circuits = [
```

Comment:
[nit]: You can just return this thing instead of defining a variable and then returning.